tests/data/attributes/operator/test_operator.py

```diff
@@ -0,0 +1,158 @@
+
+hamiltonians = [
+    qml.Hamiltonian(*args)
+    for args in [
+        (
+            [
+                1.0,
+            ],
+            (qml.Hermitian(H_TWO_QUBITS, [0, 1]),),
+        ),
+        ((-0.8,), (qml.PauliZ(0),)),
+        ((0.6,), (qml.PauliX(0) @ qml.PauliX(1),)),
+        ((0.5, -1.6), (qml.PauliX(0), qml.PauliY(1))),
+        ((0.5, -1.6), (qml.PauliX(1), qml.PauliY(1))),
+        ((0.5, -1.6), (qml.PauliX("a"), qml.PauliY("b"))),
+        ((1.1, -0.4, 0.333), (qml.PauliX(0), qml.Hermitian(H_ONE_QUBIT, 2), qml.PauliZ(2))),
+        ((-0.4, 0.15), (qml.Hermitian(H_TWO_QUBITS, [0, 2]), qml.PauliZ(1))),
+        ([1.5, 2.0], [qml.PauliZ(0), qml.PauliY(2)]),
+        (np.array([-0.1, 0.5]), [qml.Hermitian(H_TWO_QUBITS, [0, 1]), qml.PauliY(0)]),
+        ((0.5, 1.2), (qml.PauliX(0), qml.PauliX(0) @ qml.PauliX(1))),
+        ((0.5 + 1.2j, 1.2 + 0.5j), (qml.PauliX(0), qml.PauliY(1))),
+        ((0.7 + 0j, 0 + 1.3j), (qml.PauliX(0), qml.PauliY(1))),
+    ]
+]
+
+tensors = [Tensor(qml.PauliX(1), qml.PauliY(2))]
+
+
+@pytest.mark.parametrize("obs_in", [*hermitian_ops, *pauli_ops, *identity, *hamiltonians, *tensors])
+class TestDatasetOperatorObservable:
+    """Tests serializing Observable operators using the ``compare()`` method."""
+
+    def test_value_init(self, obs_in):
+        """Test that a DatasetOperator can be value-initialized
+        from an observable, and that the deserialized operator
+        is equivalent."""
+        dset_op = DatasetOperator(obs_in)
+
+        assert dset_op.info["type_id"] == "operator"
+        assert dset_op.info["py_type"] == get_type_str(type(obs_in))
+
+        obs_out = dset_op.get_value()
+        assert repr(obs_out) == repr(obs_in)
+        assert obs_in.compare(obs_out)
+
+    def test_bind_init(self, obs_in):
+        """Test that DatasetOperator can be initialized from a HDF5 group
+        that contains a operator attribute."""
+        bind = DatasetOperator(obs_in).bind
+
+        dset_op = DatasetOperator(bind=bind)
+
+        assert dset_op.info["type_id"] == "operator"
+        assert dset_op.info["py_type"] == get_type_str(type(obs_in))
+
+        obs_out = dset_op.get_value()
+        assert repr(obs_out) == repr(obs_in)
+        assert obs_in.compare(obs_out)
+
+
+class TestDatasetOperator:
+    @pytest.mark.parametrize(
+        "op_in", [qml.RX(1.1, 0), qml.FermionicSWAP(1.3, [1, "a"]), qml.Toffoli([1, "a", None])]
+    )
+    def test_value_init(self, op_in):
+        """Test that a DatasetOperator can be value-initialized
+        from an operator, and that the deserialized operator
+        is equivalent."""
+        dset_op = DatasetOperator(op_in)
+
+        assert dset_op.info["type_id"] == "operator"
+        assert dset_op.info["py_type"] == get_type_str(type(op_in))
+
+        op_out = dset_op.get_value()
+        assert repr(op_out) == repr(op_in)
+        assert op_in.data == op_out.data
+
+    def test_value_init_not_supported(self):
```

Comment:
Does this test is not being executed with any non-supported gates?

Reply:
this is just testing the general case